book support

ichvii opened the pull request:
If blueprint book is > 0, it applies every function except eject to the blueprint at position blueprint book in the book. Added ejectblueprintBook for book =-1. 

It does not add empty blueprints or rename blueprints.

Edit: I did very brief testing and at the very least the deploy feature seems to work.

ichvii committed:
Update recipes.lua

added book to recipe

ichvii committed:
Update knownsignals.lua

Added blueprint book signal

ichvii committed:
Update control.lua

ichvii committed:
Update knownsignals.lua

ichvii committed:
Update control.lua

ichvii committed:
Update control.lua

ichvii committed:
Update control.lua

justarandomgeek commented:
I generally like this, but I want to think about it a little more as I work out the blueprint editing commands before I merge it. Hopefully before the end of the week!

ichvii committed:
Update control.lua

Added a check, if a book is present. If there is no book, it will try to use the blueprint instead.
